components/director/internal/domain/formation/service.go

```diff
@@ -1758,6 +1761,21 @@ func (s *service) isValidRuntimeType(ctx context.Context, tnt string, runtimeID
 	return nil
 }
 
+func (s *service) isValidApplication(ctx context.Context, tnt string, applicationID string) error {
+	application, err := s.applicationRepository.GetByID(ctx, tnt, applicationID)
+	if err != nil {
+		return errors.Wrapf(err, "while getting application with ID %q", applicationID)
+	}
+
+	if !application.Ready {
+		return apperrors.NewInvalidOperationError(fmt.Sprintf("application with ID %q is not ready", applicationID))
+	}
+	if application.DeletedAt != nil {
+		return apperrors.NewInvalidOperationError(fmt.Sprintf("application with ID %q is currently being deleted", applicationID))
+	}
```

Comment:
Maybe we can switch the checks and check first for the deleted at and then for the ready as those properties are correlated anyways. The Delete At timestamps gives more information about why the system is not available and if the deletedAt is set the Ready will always be false.